cpp/src/uuid.cpp

```diff
@@ -42,15 +44,17 @@ namespace {
 
 // Seed the random number generated once at startup.
 struct seed {
```

Comment:
Threadsafety, I almost forgot! The std:: random stuff is not threadsafe, so maybe it should be declared thread_local as that would be easiest.

Reply:
(assuming that there is not something else which prevents concurrent use; I think there isn't)

Reply:
I use thread_local here, this is my first time using it, and I couldn't find many examples that include both cpp and hpp files. Can you please take a look?